### PR TITLE
Kong cert order

### PR DESCRIPTION
**Pull request by chenjinxuan**
image
In k8s, when the TLS secret of different namespaces is the same and the createtime is the same, the result is always synchronized configuration to Kong

### CODE REVIEW

**Comment by mflendrich** [comment]
Thank you @chenjinxuan for your contribution. I will look at your PR tomorrow.

File: .idea/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Default ignored files
```
Comment (mflendrich) [inline .idea/.gitignore:1]: Please remove the .idea directory from the PR.

**Review by mflendrich — COMMENTED** [review]
Thank you for your contribution! Apologies for getting back to you with a delay.

Also, please change the PR to target the `next` branch instead of `main`.

File: internal/ingress/controller/parser/parser.go [hunk 2]
```diff
@@ -1351,7 +1351,9 @@ func (p *Parser) getCerts(secretsToSNIs map[string][]string) []Certificate {
 				CreationTimestamp: secret.CreationTimestamp,
 			}
 		} else {
-			if kongCert.CreationTimestamp.After(secret.CreationTimestamp.Time) {
+			if kongCert.CreationTimestamp.After(secret.CreationTimestamp.Time) ||
+				(kongCert.CreationTimestamp.Equal(&secret.CreationTimestamp) &&
+					*kongCert.cert.ID > string(secret.UID)) {
```
Comment (mflendrich) [inline internal/ingress/controller/parser/parser.go:1356]: Also, please add a test for this condition (under `func TestParserSecret`)

Comment (chenjinxuan) [reply internal/ingress/controller/parser/parser.go:1356]: ok.
A new PR has been committed to the next branch

**Pull request closed without merging**